### PR TITLE
feat(ratelimiter): Support disabling rate limiting

### PR DESCRIPTION
**Pull request by tmessi**
Rate limiting can be disabled via the controller config.

### CODE REVIEW

File: internal/daemon/controller/controller.go [hunk 1]
```diff
@@ -650,27 +656,40 @@ func (c *Controller) getRateLimiter() *rate.Limiter {
 // error is returned. Otherwise the rate.Limiter is replaced, and the old rate.Limiter
 // is shutdown. This means that a client's quota is effectively reset if the
 // configuration changes.
-func (c *Controller) ReloadRateLimiter(newLimitConfigs ratelimit.Configs, newMaxEntries int) error {
+func (c *Controller) ReloadRateLimiter(newLimitConfigs ratelimit.Configs, newMaxEntries int, newDisable bool) error {
```
Comment (mgaffney) [inline internal/daemon/controller/controller.go:659]: Can we update the godoc to explain the new parameter `newDisable`?

File: internal/daemon/controller/controller.go [hunk 2]
```diff
@@ -650,27 +656,40 @@ func (c *Controller) getRateLimiter() *rate.Limiter {
...
 
-	limiter, err := ratelimit.NewLimiter(ratelimits, newMaxEntries)
-	if err != nil {
-		return err
+		limiter, err = ratelimit.NewLimiter(ratelimits, newMaxEntries)
+		if err != nil {
+			return err
+		}
+
+		ratelimit.WriteLimitsSysEvent(c.baseContext, ratelimits, newMaxEntries)
```
Comment (mgaffney) [inline internal/daemon/controller/controller.go:682]: Unhandled error

File: internal/cmd/commands/server/controller_ratelimit_reload_test.go [hunk 3]
```diff
@@ -126,6 +126,57 @@ listener "tcp" {
...
+}
+
+controller {
+	name = "test-controller"
+	description = "A default controller created for tests"
+	database {
+		url = "%s"
+	}
+
+	api_rate_limit_disable = true
```
Comment (mgaffney) [inline internal/cmd/commands/server/controller_ratelimit_reload_test.go:150]: What happens if there is also a `api_rate_limit` block in the config?

File: internal/cmd/config/config.go [hunk 4]
```diff
@@ -217,6 +217,7 @@ type Controller struct {
 
 	ApiRateLimits            ratelimit.Configs `hcl:"-"`
 	ApiRateLimiterMaxEntries int               `hcl:"api_rate_limit_max_entries"`
+	ApiRateLimitDisable      bool              `hcl:"api_rate_limit_disable"`
```
Comment (mgaffney) [inline internal/cmd/config/config.go:220]: Can we add tests for this?

**Review by mgaffney — CHANGES_REQUESTED** [review]

**Comment by tmessi** [comment]
Closing in favor of #4089

**Pull request closed without merging**